ThetaCloud.h

```diff
@@ -0,0 +1,45 @@
+#ifndef THETA_CLOUD_H
+#define THETA_CLOUD_H
+
+#include <map>
+#include <list>
+#include <memory>
+#include <functional>
+#include "SensorData.h"
+#include "DeviceHandlerToken.h"
+
+class ThetaCloud
+{
+public:
+    using Emit = std::function<void(const SensorData &)>;
+    using DeviceReadHandler = std::function<void(const Emit &)>;
+    using DeviceWriteHandler = std::function<void(const SensorData &, const Emit &)>;
+    ThetaCloud() :
+        dataCallback([](const SensorData &){
+    }),
+        initialized(false)
+    {
+    }
+    ~ThetaCloud()
+    {
+    }
+
+    void init();
+    void whenDataAvailable(const Emit &dataCallback);
+    DeviceHandlerTokenPtr addReadHandler(const DeviceReadHandler &handler);
+    DeviceHandlerTokenPtr addWriteHandler(const std::string &topic,
+                                          const DeviceWriteHandler &handler);
+    void write(const SensorData &data) const;
+    void emit(const SensorData &data) const;
+    void tick();
+private:
+
+    Emit dataCallback;
+    bool initialized;
+    std::list<DeviceReadHandler> deviceReadHandlers;
```

Comment:
why std::list not std::vector?

Reply:
no reason, I will change it as vectors would fit more nicely

Reply:
:heavy_check_mark:

Reply:
Turns out this was a wrong decision - I forgot that the `list` was used as iterators are not invalidated when items are added or removed from list, what is not the case in `vector`. Fortunately the tests found out pretty quickly ;)